File: java/samples/batch-mode/src/main/java/com/myscript/iink/samples/batchmode/IInkApplication.java
/*
 * Copyright (c) MyScript. All rights reserved.
 */

package com.myscript.iink.samples.batchmode;

import android.app.Application;

import com.myscript.certificate.MyCertificate;
import com.myscript.iink.Engine;

public class IInkApplication extends Application {

    private static Engine engine;

    public static synchronized Engine getEngine() {
        if (engine == null) {
            engine = Engine.create(MyCertificate.getBytes());
        }
        return engine;
    }

}

File: java/samples/batch-mode-kt/src/main/java/com/myscript/iink/samples/batchmode/IInkApplication.kt
package com.myscript.iink.samples.batchmode

import android.app.Application
import com.myscript.certificate.MyCertificate
import com.myscript.iink.Engine

class IInkApplication : Application() {


    companion object {

        private var engine: Engine? = null

        fun getEngine(): Engine? {
            if (IInkApplication.engine == null) {
                engine = Engine.create(MyCertificate.getBytes())
            }
            return engine
        }

        fun close() {
            engine?.close()
        }
    }

    override fun onTerminate() {
        close()
        super.onTerminate()
    }
}